app/models/spree/line_item_decorator.rb

```diff
@@ -117,7 +117,7 @@ def unit_value
   def sufficient_stock?
     return true if quantity == 0 # This line added
     scoper.scope(variant) # This line added
-    return true if Spree::Config[:allow_backorders]
+    return true if variant.on_demand
```

Comment:
I'm not sure we need to handle `on_demand` here since that was not done in Spree. Spree just looked at the global config here, not the the specific on_demand value. I guess on_demand is checked somewhere else. But it doesn't hurt to double-check.

It's a bit annoying that we are so far from Spree now that we can't use the StockQuantifier. We need a solution for VariantOverrides to get rid of these decorators.

Reply:
tbh, I am not sure what's the relation between stock_items and VariantOverrides (StockQuantifier). I need to look into that. This decorator needs to be looked at for sure. Probably a new issue. I'll investigate and create one if needed.

Reply:
Spree moved all stock logic from the variants into a service called StockQuantifier. But unfortunately that service needs to query the database (stock items) to do its job. VariantOverrides contain their own stock data and don't use stock items. Therefore the StockQuantifier can't be used for it.

So before we could just feed different data (count_on_hand) into Spree's stock logic. Now we have to implement our own stock logic (a copy of Spree's old logic at the moment).

Reply:
Is this a issue we have to create in the spree upgrade @mkllnk ?

Reply:
No, I don't think we need to change anything unless we discover that we have to. The current code could just work.

Reply:
> I'm not sure we need to handle `on_demand` here since that was not done in Spree. Spree just looked at the global config here, not the specific on_demand value. I guess on_demand is checked somewhere else. But it doesn't hurt to double-check.
> 
> It's a bit annoying that we are so far from Spree now that we can't use the StockQuantifier. We need a solution for VariantOverrides to get rid of these decorators.

I share @mkllnk here. Let's move it forward but keep an eye on redundant checks. I think soon we'll be familiar enough with this logic to confidently decide.